File: puzzles/medium/php/network-cabling.php
<?php
$minX = PHP_INT_MAX;
$maxX = 0;
$cableSize = 0;
$coordinates = [];

fscanf(STDIN, "%d", $nbBuildings);
for ($i = 0; $i < $nbBuildings; ++$i) {
  fscanf(STDIN, "%d %d", $x, $y);
  $coordinates[$i] = $y;
  if ($x < $minX)
    $minX = $x;
  if ($x > $maxX)
    $maxX = $x;
}

sort($coordinates);
$cableSize = $maxX - $minX; // main horizontal cable length
// To minimize the total vertical length of the cables,
// the y value of the main cable must be the median of all the y values.
$median = $coordinates[$nbBuildings / 2];
for ($i = 0; $i < count($coordinates); ++$i)
  $cableSize += abs($median - $coordinates[$i]);

echo("$cableSize\n");
?>

File: puzzles/medium/kotlin/src/network-cabling.kt
import java.util.Scanner

fun main(args : Array<String>) {
  val input = Scanner(System.`in`)
  val n = input.nextInt()
  var minX = Long.MAX_VALUE
  var maxX = 0L
  var cableSize: Long // main horizontal cable length
  val coordinates = ArrayList<Int>(n)

  for (i in 0 until n) {
    val x = input.nextInt()
    val y = input.nextInt()
    coordinates.add(y)
    if (x < minX) {
      minX = x.toLong()
    }
    if (x > maxX) {
      maxX = x.toLong()
    }
  }

  coordinates.sort()
  cableSize = maxX - minX

  // To minimize the total vertical length of the cables,
  // the y value of the main cable must be the median of all the y values.

  val median = coordinates[n / 2]
  for (i in 0 until n) {
    cableSize += Math.abs(median - coordinates[i])
  }

  println(cableSize)
}
